editor/scene/gui/margin_container_editor_plugin.cpp

```diff
@@ -0,0 +1,92 @@
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "margin_container_editor_plugin.h"
+
+#include "editor/scene/canvas_item_editor_plugin.h"
+#include "editor/themes/editor_scale.h"
+
+void MarginContainerEditorPlugin::edit(Object *p_object) {
+	margin_container = Object::cast_to<MarginContainer>(p_object);
+	CanvasItemEditor::get_singleton()->update_viewport();
+}
+
+bool MarginContainerEditorPlugin::handles(Object *p_object) const {
+	return p_object->is_class("MarginContainer");
+}
+
+void MarginContainerEditorPlugin::forward_canvas_draw_over_viewport(Control *p_viewport_control) {
+	if (margin_container == nullptr) {
+		return;
+	}
+
+	bool item_locked = margin_container->has_meta("_edit_lock_");
+
+	Transform2D xform = CanvasItemEditor::get_singleton()->get_canvas_transform() * margin_container->get_screen_transform();
+
+	// TODO: These colors are copied from CanvasItemEditor::_draw_selection.
+	// We may want to unify them somehow.
+	Color border_color = Color(1, 0.6, 0.4, 0.5);
+
+	if (item_locked) {
+		border_color = Color(0.7, 0.7, 0.7, 0.5);
+	}
+
+	int border_width = Math::round(1 * EDSCALE);
+
+	Rect2 rect = margin_container->_edit_get_rect();
+
+	int margin_left = margin_container->get_theme_constant("margin_left");
+	int margin_top = margin_container->get_theme_constant("margin_top");
+	int margin_right = margin_container->get_theme_constant("margin_right");
+	int margin_bottom = margin_container->get_theme_constant("margin_bottom");
```

Comment:
You could use `get_margin_size()` instead.

Reply:
Code-wise, it definitely looks like that - I'll give that a shot!

However, strangely, I don't see any documentation for it. https://docs.godotengine.org/en/latest/classes/class_margincontainer.html
It appears to be a public, non-overriding method, so it seems like it should be documented. Even just searching Google for it doesn't turn up anything: https://www.google.com/search?q=site%3Adocs.godotengine.org+%22get_margin_size%22

Should this be addressed in another PR, or is it intended to be internal to the engine?